update current XDCR page for active-active (#916)
couchbaselabs/docs-sync-gateway: modules/server-compatibility/pages/server-compatibility-xdcr.adoc

TIP: See: {compatibility--xref} for version compatibility information.

Both {cbm-te} and {server-xdcr--xref} (XDCR) provide for efficient and secure replication of data, using different replication technologies optimized for specific use cases.

Sync Gateway provides different replication approaches depending on your version and use case requirements:

* **Sync Gateway 4.0+**: Bi-directional XDCR with version vectors enables active-active mobile clusters
* **Sync Gateway < 4.0**: Inter-Sync Gateway Replication for active-active, unidirectional XDCR for specific scenarios

Here we provide details on how XDCR feature relates to the Couchbase Mobile ecosystem.

[#lbl-bidirectional-xdcr]

> == Bi-directional XDCR

Reviewer: The headings on this page are a little weird.

<img width="263" height="415" alt="Screenshot 2025-09-18 at 18 59 35" src="https://github.com/user-attachments/assets/95a9f6c2-71ad-4743-a71d-20b46fc8a10d" />

Feels like `Bi-directional XDCR` needs to be underneath `Cross Data Center Replication (XDCR)` but the ISGR section being above is preventing that.

Maybe the ISGR section can be moved way down the page now active-active XDCR is supported?

Author: I've updated this to reflect this , I've also updated the preview URL on the description above.

Sync Gateway 4.0 introduces support for bi-directional XDCR between active mobile clusters through an architectural change from revision trees to version vectors.

This enables active-active deployments where multiple clusters can process writes simultaneously while maintaining data consistency across geographically distributed datacenters.
Both clusters process reads and writes from mobile clients simultaneously.

Sync Gateway 4.0 uses automatic conflict resolution with a `Last-Write-Wins` method and hybrid logical clocks by default.
Clients can switch seamlessly between clusters without data loss.
This provides zero-downtime disaster recovery with immediate fail-over capabilities.

> TIP: For detailed technical implementation and configuration requirements, see: [Bi-directional XDCR with Mobile Clusters].

Reviewer: Dunno if this link is for the XDCR docs? ~They have good instructions for active-active setup.~

~https://docs.couchbase.com/server/current/manage/manage-xdcr/create-xdcr-replication.html#create-an-xdcr-replication-with-mobile-as-active~

Edit: I maybe found a better page on XDCR for this

https://docs.couchbase.com/server/current/learn/clusters-and-availability/xdcr-active-active-sgw.html#xdcr-active-active-sgw-prerequisites

See: {sync-inter-syncgateway-overview--xref} for complete implementation details.

Inter-Sync Gateway replication provides bi-directional read/write replications that verify:

* Sync Gateway observes cluster-specific security by invoking the appropriate Sync Function
* Sync Gateway maintains the integrity of security history with historical access rules held in metadata
* All clusters use a consistent Revision Id, allowing clients to identify a revision regardless of cluster location
* Cluster-specific `_sync` documents remain local and are not replicated to other mobile clusters

[[icr-active-mobile]]
.Inter-Sync Gateway Active-to-Active Replication
image::ROOT:{image-icr-active-mobile-sync}[,std-image-size]

TIP: To upgrade to Sync Gateway 4.0 and enable bi-directional XDCR for active-active mobile clusters, see: xref:ROOT:upgrading.adoc[Upgrading Sync Gateway].

[#lbl-xdcr]
== Cross Data Center Replication (XDCR)

> Couchbase Server's Cross Data Center Replication (xref:server:manage:manage-xdcr/xdcr-management-overview.adoc[XDCR]) replicates all of Sync Gateway's metadata (`_sync xattr`) along with associated documents, making it suitable for various mobile cluster replication scenarios.

Reviewer: Actually with 4.0 and active-active XDCR replication we _don't_ replicate `_sync` across to both sides - both sides maintain independent sync metadata data and only share version information and global metadata (like attachments).

```suggestion
Couchbase Server's Cross Data Center Replication (xref:server:manage:manage-xdcr/xdcr-management-overview.adoc[XDCR]) replicates necessary Sync Gateway metadata along with associated documents, making it suitable for various mobile cluster replication scenarios.
```
